src/data/set/intervals/basic.lean

```diff
@@ -1153,6 +1159,9 @@ by { ext x, simp [Ioi] }
 by rw [← Ioi_inter_Iic, inter_assoc, inter_comm, inter_assoc, Ioi_inter_Ioi, inter_comm,
   Ioi_inter_Iic, sup_comm]
 
+@[simp] lemma Ioi_inter_Ioc [is_total α (≤)] {a b c : α} : Ioi a ∩ Ioc b c = Ioc (a ⊔ b) c :=
```

Comment:
This is a weird phrasing. Shouldn't you rather use `linear_order α` and `max a b`? And change the above lemmas while you're at it.